probe-rs/src/session.rs

```diff
@@ -334,14 +336,9 @@ impl Session {
             configured_trace_sink: None,
         };
 
-        {
-            // Todo: Add multicore support. How to deal with any cores that are not active and won't respond?
-            let mut core = session.core(0)?;
-
-            core.halt(Duration::from_millis(100))?;
-        }
-
-        sequence_handle.on_connect(session.get_xtensa_interface()?)?;
+        session.halted_access(|sess| {
```

Comment:
This seems to break Xtensa, and I don't really know why. I guess there are two material changes:

- The core that was running is now resumed after
- If the second core was running it will also be halted

Neither of which explain this behaviour :thinking:

Reply:
I believe currently the BreakIn/BreakOut configuration in the debug module mean that the cores halt more or less simultaneously, which shouldn't make a difference here. I wonder if issueing a `debug_int_break` in `Xdm/schedule_halt` actually causes an (undocumented?) exception here? Maybe we should check that the core is actually running before we halt it.